### PR TITLE
Adjusted the Search Query to be Case Sensitive.

### PR DESCRIPTION
**Pull request by tonypartridge**
More work is needed, StringHelper::substr_replace doesn't find a result?? and the description highlighting is always forcing lower on the search and match replacing with the keyword.

Pull Request for Issue #17241 

### Testing Instructions
Create an article with title: Article and Description: Article

Search: article, both article and Article are found. 

### Expected result

The exact string is found not both upper and lower case.

### Actual result

All cases are found.

### CODE REVIEW

File: components/com_search/views/search/view.html.php [hunk 1]
```diff
@@ -258,16 +260,22 @@ public function display($tpl = null)
...
-				$result->title   = StringHelper::str_ireplace($needle, $hl1 . $needle . $hl2, htmlspecialchars($result->title, ENT_COMPAT, 'UTF-8'));
+				$result->title   = StringHelper::substr_replace(
+					$needle,
+					$hl1 . $needle . $hl2,
+					htmlspecialchars(
+						$result->title, ENT_COMPAT, 'UTF-8'
+					)
+				);
+
+				var_dump($result->text);die;
```
Comment (zero-24) [inline components/com_search/views/search/view.html.php]: this looks like a mistake? ;)

Comment (zero-24) [reply components/com_search/views/search/view.html.php]: and btw. also results into the drone error :p http://213.160.72.75/joomla/joomla-cms/7273

**Comment by tonypartridge** [comment]
thanks @zero-24  I pushed before last commit lol.

**Comment by zero-24** [comment]
thanks drone is happy now ;)

**Pull request closed without merging**